providers/job.rb

```diff
@@ -0,0 +1,70 @@
+use_inline_resources
+
+def whyrun_supported?
+  true
+end
+
+def load_current_resource
+  @current_resource = Chef::Resource::RundeckServerJob.new(@new_resource.name)
+  @current_resource.project(@new_resource.project)
+end
+
+action :create do
+  require 'rundeck'
+  require 'yaml'
+
+  client = Rundeck.client(endpoint: @new_resource.endpoint, api_token: @new_resource.api_token)
+
+  job = find_job(client, @current_resource.project, @current_resource.name)
+  updated_job = @new_resource.config.dup
+  updated_job[:uuid] = job[:uuid] if job
+
+  action_name = 'create'
+  action_name = 'update' if job
+
+  if job.nil? || job.to_hash != updated_job
+    # hydrate updated_job
+    updated_job[:name] = @current_resource.name
+    updated_job[:project] = @current_resource.project
+
+    # yamlize update_job
+    job_yaml = updated_job.to_yaml.gsub(/^---\n/, '- ').gsub(/\n/, "\n  ")
```

Comment:
because:
-  `to_yaml` does not output the kind of yaml that the api want (first gsub)
- the outputed yaml is a sub part of the definitive yaml, so we need to indent it by 2 chars